client/src/unifyfs.c

```diff
@@ -924,7 +924,8 @@ int unifyfs_fid_create_file(const char* path)
     unifyfs_filelist[fid].in_use = 1;
 
     /* copy file name into slot */
-    strncpy((void*)&unifyfs_filelist[fid].filename, path, UNIFYFS_MAX_FILENAME);
+    strncpy((void*)&unifyfs_filelist[fid].filename, path,
+             UNIFYFS_MAX_FILENAME - 1);
```

Comment:
This is to resolve the build error that showed up on GCC 10.1 which I believe is the same error in #609.
Want to check that this is the solution we want to use here.

Reply:
In other places where newer GCC was complaining about buffer sizes, Tony switched from `strncpy` to `strlcpy` implemented in `common/src/unifyfs_misc.[ch]`

Reply:
Switched to `strlcpy`.